src/components/LinkToRecord/LinkToRecord.test.ts

```diff
@@ -0,0 +1,33 @@
+import { describe, expect, test } from 'vitest';
+import { renderToFragment } from '@lib/renderer';
+import type { RecordRoute } from '@lib/routing';
+import LinkToRecord, { type Props as LinkToRecordProps } from './LinkToRecord.astro';
+
+const record = {
+  title: 'Homepage',
+  __typename: 'HomePageRecord',
+} satisfies RecordRoute;
+
+const props = {
+  record
+} satisfies LinkToRecordProps;
+
+describe('LinkToRecord', () => {
+  test('Component is rendered', async () => {
+    const fragment = await renderToFragment<LinkToRecordProps>(
+      LinkToRecord,
+      { props }
+    );
+    expect(fragment).toBeDefined();
+  });
+
+  test('renders text content without trailing whitespace ', async () => {
+    const fragment = await renderToFragment<LinkToRecordProps>(LinkToRecord, {
+      props,
+      slots: {
+        default: 'spacing\n',
+      },
+    });
+    expect(fragment.querySelector('a')?.textContent).toBe('spacing');
+  });
```

Comment:
Is this something we want to test in this file? Trimming is done in the `Link` component and if we change this implementation of the Link component, we would not want this to fail (I think). Seems to me that this is a implementation detail that we test in `Link.test.ts`.

Reply:
You're right, but since I've spent some time diving through the whole stack of components to find that issue, I felt it wouldn't hurt to guard against this a bit more vociferously